JavaScript/Topics/7.Execution_Context/README.md

```diff
@@ -1,6 +1,56 @@
 # Execution Context in JavaScript
 
-When running code, **Execution Context** gets created by our JS engine, which does all the magic in the background. Here's how!
+Before diving deep into what the execution context is we have to first understand what the javascript engine is because it plays a crucial role and performs a lot of tasks before the role of execution context came.
+
+# What is Javascript Engine?
+Javascript Engine is nothing but a program that executes our javascript code. Every browser has its own Javascript engine but probably the most well-known engine is Google's V8.
+Javascript Engine has actually two components:
+# 1. Callstack
+ The CallStack is the place where our code is actually executed using execution context. Inside the CallStack various execution contexts are stacked on the top of one another and gets popped out from the stack whenever the entire code present inside the execution context is executed.
+# 2. Heap
+On the other hand, Heap is a unstructured memory pool which stores all the objects that our application needs.
+
+# Task of JS Engine
+Before starting out the task of execution context or we say before execution context came into picture, the very first step which is performed is to actually send the piece of javascript code into the javascript engine.
```

Comment:
Well done @Shishiranjan , just make it the info in bullet points so it'll better to read.

Reply:
Sure will do that.

Reply:
@s4-2 Maam Done, please review it.